src/esp/sim/ReplayBatchRenderer.cpp

```diff
@@ -168,8 +362,179 @@ scene::SceneNode* ReplayBatchRenderer::loadAndCreateRenderAssetInstance(
   const auto& env = envs_[envIndex];
   // perf todo: avoid dynamic mem alloc
   std::vector<int> tempIDs{env.sceneID_, env.semanticSceneID_};
-  return resourceManager_->loadAndCreateRenderAssetInstance(
+  auto* node = resourceManager_->loadAndCreateRenderAssetInstance(
       assetInfo, creation, sceneManager_.get(), tempIDs);
+  return reinterpret_cast<gfx::replay::NodeHandle>(node);
+}
+
+ReplayBatchRenderer::ReplayBatchRenderer(
+    const ReplayRendererConfiguration& cfg) {
+  CORRADE_ASSERT(cfg.sensorSpecifications.size() == 1,
+                 "ReplayBatchRenderer: expecting exactly one sensor", );
+  const auto& sensor = static_cast<esp::sensor::CameraSensorSpec&>(
+      *cfg.sensorSpecifications.front());
+
+  gfx_batch::RendererConfiguration configuration;
+  configuration.setTileSizeCount(Mn::Vector2i{sensor.resolution}.flipped(),
+                                 environmentGridSize(cfg.numEnvironments));
+  if ((standalone_ = cfg.standalone))
+    renderer_.emplace<gfx_batch::RendererStandalone>(
+        configuration, gfx_batch::RendererStandaloneConfiguration{});
```

Comment:
We can carry the GPU ID here:
```
renderer_.emplace<gfx_batch::RendererStandalone>(
        configuration, gfx_batch::RendererStandaloneConfiguration{}
          .setCudaDevice(cfg.gpuDeviceId));
```

Reply:
Good point, I'll add this -- seems important.

Reply:
Ah well, the default `ReplayRendererConfiguration::gpuDeviceId` is `0`, but `0` means the first CUDA device -- i.e., if you have no NV cards (or the NV card disconnected as in my case), it would fail to create the context.

The default should be `-1` I think, i.e. picking whatever the primary GPU is, so it's not restricted to work only on NVidias. Can I change the default to `-1`?

Reply:
Oh that's right. `-1` by default makes sense then.

Reply:
Not sure if I want to go through the whole CI process again -- we want to merge ASAP -- could you add this in the followup PR? :)

Reply:
Sounds good!

Reply:
The default `ReplayRendererConfiguration::gpuDeviceId` is `0` because the default `SimulatorConfiguration::gpuDeviceId` is `0`. @0mdc , if you're making a change, consider making it in both places.